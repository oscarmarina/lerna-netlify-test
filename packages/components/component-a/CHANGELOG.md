# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [1.0.2](https://github.com/oscarmarina/lerna-netlify-test/compare/@blockquote-playground/component-a@1.0.1...@blockquote-playground/component-a@1.0.2) (2025-10-26)

**Note:** Version bump only for package @blockquote-playground/component-a





## 1.0.1 (2025-10-01)


### Bug Fixes

* **component-a:** update counter ([1fe3db5](https://github.com/oscarmarina/lerna-netlify-test/commit/1fe3db580de41c158430f2893d416afec6531fec))



# 0.0.0 (2025-10-01)





# 1.0.0 (2025-09-30)

**Note:** Version bump only for package @blockquote-playground/component-a
