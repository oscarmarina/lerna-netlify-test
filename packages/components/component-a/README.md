![Lit](https://img.shields.io/badge/lit-3.0.0-blue.svg)

## `<component-a>`
An example element.


### `src/ComponentA.js`:

#### class: `ComponentA`, `component-a`

##### Fields

| Name      | Privacy | Type     | Default       | Description                                      | Inherited From |
| --------- | ------- | -------- | ------------- | ------------------------------------------------ | -------------- |
| `heading` | public  | `string` | `'Hey there'` | The heading to say "Hello" to.                   |                |
| `counter` | public  | `number` | `1`           | The number of times the button has been clicked. |                |

##### Methods

| Name       | Privacy | Description        | Parameters        | Return   | Inherited From |
| ---------- | ------- | ------------------ | ----------------- | -------- | -------------- |
| `sayHello` |         | Formats a greeting | `heading: string` | `string` |                |

##### Events

| Name            | Type          | Description                      | Inherited From |
| --------------- | ------------- | -------------------------------- | -------------- |
| `counterchange` | `CustomEvent` | Indicates when the count changes |                |

##### Attributes

| Name      | Field   | Inherited From |
| --------- | ------- | -------------- |
| `heading` | heading |                |
| `counter` | counter |                |

##### Slots

| Name | Description             |
| ---- | ----------------------- |
|      | This element has a slot |

<details><summary>Private API</summary>

##### Methods

| Name       | Privacy | Description | Parameters | Return | Inherited From |
| ---------- | ------- | ----------- | ---------- | ------ | -------------- |
| `#onClick` | private |             |            |        |                |

</details>

<hr/>

#### Exports

| Kind | Name         | Declaration | Module            | Package |
| ---- | ------------ | ----------- | ----------------- | ------- |
| `js` | `ComponentA` | ComponentA  | src/ComponentA.js |         |

### `src/index.js`:

#### Exports

| Kind | Name         | Declaration | Module          | Package |
| ---- | ------------ | ----------- | --------------- | ------- |
| `js` | `ComponentA` | ComponentA  | ./ComponentA.js |         |

### `src/define/component-a.js`:

#### Exports

| Kind                        | Name          | Declaration | Module             | Package |
| --------------------------- | ------------- | ----------- | ------------------ | ------- |
| `custom-element-definition` | `component-a` | ComponentA  | /src/ComponentA.js |         |

### `src/styles/component-a-styles.css.js`:

#### Variables

| Name     | Description | Type |
| -------- | ----------- | ---- |
| `styles` |             |      |

<hr/>

#### Exports

| Kind | Name     | Declaration | Module                               | Package |
| ---- | -------- | ----------- | ------------------------------------ | ------- |
| `js` | `styles` | styles      | src/styles/component-a-styles.css.js |         |
