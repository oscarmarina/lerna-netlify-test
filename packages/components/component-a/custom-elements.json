{
  "schemaVersion": "1.0.0",
  "readme": "",
  "modules": [
    {
      "kind": "javascript-module",
      "path": "src/ComponentA.js",
      "declarations": [
        {
          "kind": "class",
          "description": "![Lit](https://img.shields.io/badge/lit-3.0.0-blue.svg)\n\n## `<component-a>`\nAn example element.",
          "name": "ComponentA",
          "slots": [
            {
              "description": "This element has a slot",
              "name": ""
            }
          ],
          "members": [
            {
              "kind": "method",
              "name": "#onClick",
              "privacy": "private"
            },
            {
              "kind": "method",
              "name": "sayHello",
              "parameters": [
                {
                  "name": "heading",
                  "description": "The heading to say \"Hello\" to",
                  "type": {
                    "text": "string"
                  }
                }
              ],
              "description": "Formats a greeting",
              "return": {
                "type": {
                  "text": "string"
                }
              }
            },
            {
              "kind": "field",
              "name": "heading",
              "privacy": "public",
              "type": {
                "text": "string"
              },
              "description": "The heading to say \"Hello\" to.",
              "default": "'Hey there'",
              "attribute": "heading"
            },
            {
              "kind": "field",
              "name": "counter",
              "privacy": "public",
              "type": {
                "text": "number"
              },
              "description": "The number of times the button has been clicked.",
              "default": "1",
              "attribute": "counter"
            }
          ],
          "events": [
            {
              "name": "counterchange",
              "type": {
                "text": "CustomEvent"
              },
              "description": "Indicates when the count changes"
            }
          ],
          "attributes": [
            {
              "name": "heading",
              "type": {
                "text": "string"
              },
              "description": "The heading to say \"Hello\" to.",
              "default": "'Hey there'",
              "fieldName": "heading"
            },
            {
              "name": "counter",
              "type": {
                "text": "number"
              },
              "description": "The number of times the button has been clicked.",
              "default": "1",
              "fieldName": "counter"
            }
          ],
          "superclass": {
            "name": "LitElement",
            "package": "lit"
          },
          "tagName": "component-a",
          "customElement": true
        }
      ],
      "exports": [
        {
          "kind": "js",
          "name": "ComponentA",
          "declaration": {
            "name": "ComponentA",
            "module": "src/ComponentA.js"
          }
        }
      ]
    },
    {
      "kind": "javascript-module",
      "path": "src/index.js",
      "declarations": [],
      "exports": [
        {
          "kind": "js",
          "name": "ComponentA",
          "declaration": {
            "name": "ComponentA",
            "module": "./ComponentA.js"
          }
        }
      ]
    },
    {
      "kind": "javascript-module",
      "path": "src/define/component-a.js",
      "declarations": [],
      "exports": [
        {
          "kind": "custom-element-definition",
          "name": "component-a",
          "declaration": {
            "name": "ComponentA",
            "module": "/src/ComponentA.js"
          }
        }
      ]
    },
    {
      "kind": "javascript-module",
      "path": "src/styles/component-a-styles.css.js",
      "declarations": [
        {
          "kind": "variable",
          "name": "styles",
          "default": "css` :host { display: block; box-sizing: border-box; background-color: olive; } :host([hidden]), [hidden] { display: none !important; } *, *::before, *::after { box-sizing: inherit; } `"
        }
      ],
      "exports": [
        {
          "kind": "js",
          "name": "styles",
          "declaration": {
            "name": "styles",
            "module": "src/styles/component-a-styles.css.js"
          }
        }
      ]
    }
  ]
}
