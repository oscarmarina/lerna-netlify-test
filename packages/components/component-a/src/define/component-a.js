import {ComponentA} from '../ComponentA.js';

window.customElements.define('component-a', ComponentA);
