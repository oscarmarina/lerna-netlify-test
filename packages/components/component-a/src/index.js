export {ComponentA} from './ComponentA.js';
