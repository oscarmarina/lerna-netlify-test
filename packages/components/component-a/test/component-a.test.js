import {
  suite,
  test,
  assert,
  expect,
  beforeAll,
  afterAll,
  beforeEach,
  afterEach,
  vi,
  chai,
} from 'vitest';
import {getElementLocatorSelectors} from '@vitest/browser/utils';
import {html} from 'lit';
import {chaiA11yAxe} from 'chai-a11y-axe';
import {fixture, fixtureCleanup} from '@open-wc/testing-helpers';
import {getDiffableHTML} from '@open-wc/semantic-dom-diff/get-diffable-html.js';
import '../src/define/component-a.js';

chai.use(chaiA11yAxe);

suite('ComponentA', () => {
  /**
   * @type {import('../src/index').ComponentA}
   */
  let el;
  let elShadowRoot;
  let elLocator;

  suite('Semantic Dom and a11y', () => {
    beforeAll(async () => {
      el = await fixture(html`<component-a>light-dom</component-a>`);
      elShadowRoot = el?.shadowRoot?.innerHTML;
      elLocator = getElementLocatorSelectors(el);
    });

    afterAll(() => {
      fixtureCleanup();
    });

    test('has a default heading "Hey there" and counter 1', () => {
      const button = elLocator.getByText('Counter: 1').query();
      const heading = elLocator.getByText('Hey there').query();
      assert.isOk(button);
      assert.isOk(heading);
    });

    test('SHADOW DOM - Structure test', () => {
      expect(getDiffableHTML(elShadowRoot)).toMatchSnapshot('SHADOW DOM');
    });

    test('LIGHT DOM - Structure test', () => {
      expect(getDiffableHTML(el, {ignoreAttributes: ['id']})).toMatchSnapshot('LIGHT DOM');
    });

    test('a11y', async () => {
      await assert.isAccessible(el);
    });
  });

  suite('Events ', () => {
    beforeEach(async () => {
      el = await fixture(html`<component-a></component-a>`);
      elLocator = getElementLocatorSelectors(el);
    });

    afterEach(() => {
      fixtureCleanup();
    });

    test('should increment value on click', async () => {
      const button = elLocator.getByText('Counter: 1');
      const elButton = button.query();
      await button.dblClick();
      await el.updateComplete;
      assert.include(elButton.textContent, 'Counter: 3');
    });

    test('counterchange event is dispatched', async () => {
      const spyEvent = vi.spyOn(el, 'dispatchEvent');
      const button = elLocator.getByText('Counter: 1');
      await button.click();
      await el.updateComplete;
      const calledWithCounterChange = spyEvent.mock.lastCall?.[0].type === 'counterchange';
      assert.isTrue(calledWithCounterChange);
    });
  });

  suite('Override ', () => {
    beforeAll(async () => {
      el = await fixture(html`<component-a heading="attribute heading"></component-a>`);
      elLocator = getElementLocatorSelectors(el);

      return () => {
        fixtureCleanup();
      };
    });

    test('can override the heading via attribute', () => {
      assert.equal(el.heading, 'attribute heading');
    });
  });
});
