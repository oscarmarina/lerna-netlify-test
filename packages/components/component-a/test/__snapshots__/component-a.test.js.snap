// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ComponentA > Semantic Dom and a11y > LIGHT DOM - Structure test > LIGHT DOM 1`] = `
"<component-a>
  light-dom
</component-a>
"
`;

exports[`ComponentA > Semantic Dom and a11y > SHADOW DOM - Structure test > SHADOW DOM 1`] = `
"<h1>
  Hello, Hey there!
</h1>
<button>
  Counter: 1
</button>
<hr>
<slot>
</slot>
"
`;
